include/unifex/connect_awaitable.hpp

```diff
@@ -160,24 +165,28 @@ namespace _await_cpo {
         // for the receiver to destroy the coroutine.
         co_yield [&](result_type&& result) {
               return [&] {
-                constexpr bool canCompleteWithValue = Awaitable::template value_types<count_types, single_value_type>::value > 0;
-                if constexpr (canCompleteWithValue) {
-                  using value_type = typename Awaitable::template value_types<single_type, single_value_type>;
-                  if (result.has_value()) {
-                    if constexpr (std::is_void_v<value_type>) {
-                      unifex::set_value(std::move(receiver));
-                    } else {
-                      unifex::set_value(std::move(receiver), std::move(result).value());
-                    }
-                    return;
+                constexpr size_t valueOverloadCount = Awaitable::template value_types<count_types, single_value_type>::value;
+                static_assert(valueOverloadCount <= 1);
+
+                if constexpr (valueOverloadCount == 1) {
+                  constexpr size_t valueCount = Awaitable::template value_types<type_identity_t, count_types>::value;
+                  if constexpr (valueCount == 0) {
+                    unifex::set_value(std::move(receiver));
+                  } else if constexpr (valueCount == 1) {
+                    using value_type = typename Awaitable::template value_types<single_type, single_type>::type::type;
+                    unifex::set_value(std::move(receiver), static_cast<value_type&&>(result));
+                  } else {
+                    std::apply([&](auto&&... values) {
+                      unifex::set_value(std::move(receiver), static_cast<decltype(values)>(values)...);
+                    }, (result_type&&)result);
```

Comment:
This lambda only really depends on the `Receiver` type, but due to how this code is structured it also depends incidentally on the `Awaitable` type. IMO, it's better to factor this out. Kill all the lambdas.

Reply:
The lambda has been removed.